corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/menus/menu_controller.js

```diff
@@ -0,0 +1,55 @@
+/*global FormplayerFrontend */
+
+FormplayerFrontend.module("AppSelect.MenuList", function (MenuList, FormplayerFrontend, Backbone, Marionette, $, _) {
+    MenuList.Controller = {
+        selectMenu: function (app_id, select_list) {
+
+            var fetchingApps = FormplayerFrontend.request("app:select:menus", app_id, select_list);
+
+            $.when(fetchingApps).done(function (options) {
+                var menuListView;
+                if (options.type === "commands") {
+                    menuListView = new MenuList.MenuListView({
+                        collection: options,
+                        title: options.title,
+                    });
+                }
+                else if (options.type === "entities") {
+                    menuListView = new MenuList.CaseListView({
+                        collection: options,
+                        title: options.title,
+                    });
+                } else{
+                    //TODO: error handle this, we didn't recognize the JSON resposne
+                }
+
+                FormplayerFrontend.regions.main.show(menuListView.render());
+            });
+        },
+
+        showDetail: function (model) {
+            var headers = model.options.model.attributes.detail.headers;
```

Comment:
is this a backbone model? can it be called as `model.get('headers')`. also the nested model.options.model is odd, what does that represent?